### PR TITLE
Cleanup rollout notebooks and povertymapping modules

### PR DESCRIPTION
**Pull request by tm-jace-peralta**
**Problem:**
- We got into unexpected issues during rollout because of differences in common and official country names (e.g. "laos":"lao democratic peoples republic"

**Initial solution**
- Cleanup `hdx.py` functions to take in the ISO country code as parameter instead of the official country name.

**Problem with initial solution**
- The functions in `hdx.py` require both the common country name and ISO code at different stages to construct the urls, and other functions dependent on it (e.g. those in `rollout_grids.py`) take in the common country name too. It can be tedious to trace the variables one by one to update this.

**Proposed solution**
- The errors from the inconsistencies are triggered from `iso3.py`, so its more straightforward and less time consuming to start cleanup from here 
- I modified `iso3.py` to read from a different [table](https://datahub.io/core/country-codes/r/country-codes.csv) that contains common country names, add code to lookup common country name and output the ISO code.
- I propose to keep `hdx.py` as is, and do the translation of ISO code to common country name in the rollout notebooks.

**Changes in rollout notebooks:**

CURRENT:
```python
# Imports
from povertymapping.iso3 import get_iso3_code
# Cell for  setting parameters
COUNTRY_OSM = "laos"  
COUNTRY_CODE= get_iso3_code(COUNTRY_OSM, code='alpha-2').lower()
```
PROPOSED:
```python
# Imports
from povertymapping.iso3 import get_region_name
# Cell for  setting parameters
COUNTRY_CODE = "la" 
COUNTRY_OSM = get_region_name(COUNTRY_CODE, code='alpha-2').lower()
```
- Consistent ordering of sections (feature generation, engineering, prediction, viz)
- Cleaned up markdowns for better rendering in the docs page

### CODE REVIEW

**Comment by alronlam** [comment]
Some questions:

1. So essentially, the main change is that instead of the user supplying the "human-friendly" OSM region name and the code inferring the ISO code from it, we reverse it? That is, user supplies the 2-letter ISO country code, and then we infer the OSM name from it? 

2. Have you checked that all the rollout notebooks run successfully with these changes? I also don't see the notebook changes in the PR yet.

3. Does this change impact any notebooks outside from the rollout ones?

File: povertymapping/iso3.py [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
 
 @lru_cache(maxsize=None)
 def get_iso3_codes():
     iso3_df = pd.read_csv(ISO3_URL)
-    iso3_lookup = {item['name'].lower(): item for item in iso3_df.to_dict('records')}
+    iso3_df.columns = [col.lower().replace('_','-').replace(' ','-') for col in iso3_df.columns]
+    iso3_lookup = iso3_df[['iso3166-1-alpha-3','cldr-display-name']].apply(lambda x: x.astype(str).str.lower())
+    iso3_lookup['cldr-display-name'] = iso3_lookup['cldr-display-name'].str.replace(' ','-')
+    iso3_lookup['iso-alpha-2'] = iso3_lookup['iso3166-1-alpha-3'].apply(lambda x: x[:-1])
+    iso3_lookup.columns = ['iso-alpha-3','common-name','iso-alpha-2']
```
Comment (alronlam) [inline povertymapping/iso3.py:17]: where does common name come from? can't seem to find it in the CSV file?

File: povertymapping/iso3.py [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
-
+ISO3_URL = "https://datahub.io/core/country-codes/r/country-codes.csv"
 
 @lru_cache(maxsize=None)
 def get_iso3_codes():
     iso3_df = pd.read_csv(ISO3_URL)
-    iso3_lookup = {item['name'].lower(): item for item in iso3_df.to_dict('records')}
+    iso3_df.columns = [col.lower().replace('_','-').replace(' ','-') for col in iso3_df.columns]
+    iso3_lookup = iso3_df[['iso3166-1-alpha-3','cldr-display-name']].apply(lambda x: x.astype(str).str.lower())
+    iso3_lookup['cldr-display-name'] = iso3_lookup['cldr-display-name'].str.replace(' ','-')
```
Comment (alronlam) [inline povertymapping/iso3.py:15]: where is this being used? it seems like at the end we just use iso-alpha-3/2 and common-name?

File: povertymapping/iso3.py (hunk 1 above)
Comment (tm-jace-peralta) [reply povertymapping/iso3.py:17]: original column name is `cldr-display-name`, just renamed it for readability

**Comment by alronlam** [comment]
Finished the minor bug fixes, refactors, and re-running all notebooks with @tm-jc-nacpil. All models and output data has been refreshed in the [public GDrive folder](https://drive.google.com/drive/u/0/folders/1QX0xJc6MHxY7dzIsVMDm5TH0F-NwXhBW).

Tests are fixed and updated too.

**Review by alronlam — APPROVED** [review]

**Pull request merged**